sql/src/main/java/org/apache/druid/sql/SqlRowTransformer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.druid.sql.calcite.planner.Calcites;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.ISODateTimeFormat;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class transforms the values of TIMESTAMP or DATE type for sql query results.
+ * The transformation is required only when the sql query is submitted to {@link org.apache.druid.sql.http.SqlResource}.
+ */
+public class SqlRowTransformer
```

Comment:
nice :+1: I wonder if someday we should consolidate the transformation that is done here, with the stuff [`QueryMaker` is doing](https://github.com/apache/druid/blob/master/sql/src/main/java/org/apache/druid/sql/calcite/rel/QueryMaker.java#L165), like maybe this transformer could be pushed all the way down, and maybe handle [the special stuff it is doing for JDBC there the same way with a transformer instead]( https://github.com/apache/druid/blob/master/sql/src/main/java/org/apache/druid/sql/calcite/rel/QueryMaker.java#L332)

Reply:
Yeah I think we should consolidate them someday. I didn't do it in this PR to avoid invasive refactoring.